vcd/datasource_vcd_external_network_v2.go

```diff
@@ -0,0 +1,89 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceVcdExternalNetworkV2() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceVcdExternalNetworkV2Read,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ip_scope": &schema.Schema{
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set of IP scopes for the network",
+				Elem:        networkV2IpScope,
+			},
+			"vsphere_network": &schema.Schema{
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A list of port groups that back this network. Each referenced DV_PORTGROUP or NETWORK must exist on a vCenter server registered with the system.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vcenter_id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The vCenter server ID",
+						},
+						"portgroup_id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The portgroup ID",
+						},
+					},
+				},
+			},
+			"nsxt_network": &schema.Schema{
```

Comment:
What is the reason to group `nsxt_manager_id` and `nsxt_tier0_router_id` into a `TypeSet` with `MaxItems: 1`? Curious, is there a benefit compared to leaving these parameters flat. The reason I'm asking, because we had hit issues with the `TypeSet` in the past, so just would like to get confirmation that this won't be the case here :)

Reply:
The reason was that adjacent block vsphere_network can have more than one and is a TypeSet. I can change this one to TypeList. We can switch back if it allows to specify more than 1 in future.

Reply:
Ah! I haven't realized it relates to NSX-V's `vsphere_network` and that more than one can be defined.
It's OK, if you see no technical issues with `TypeSet`, let's leave it.